webapp/src/ts/services/target-aggregates.service.ts

```diff
@@ -71,18 +86,24 @@ export class TargetAggregatesService {
       });
   }
 
-  private fetchLatestTargetDoc(settings, contactUuid) {
-    const tag = this.getCurrentIntervalTag(settings);
+  private async fetchTargetDocs(settings, contactUuid) {
+    const tagStart = this.getCurrentIntervalTag(settings);
+    const tagEnd = this.getOldIntervalTag(settings);
     const opts = {
-      start_key: `target~${tag}~${contactUuid}~`,
-      end_key: `target~${tag}~${contactUuid}~\ufff0`,
-      include_docs: true
+      start_key: `target~${tagStart}~${contactUuid}~\ufff0`,
+      end_key: `target~${tagEnd}~${contactUuid}~`,
+      descending: true,
     };
+    const bulkTargetDocs = await this.dbService.get().allDocs(opts);
+    const targetDocsIds = bulkTargetDocs.rows
+      .filter(row => row.id.includes(`~${contactUuid}~`))
```

Comment:
How could the query return a doc with an ID that didn't include the contactUuid? ie: what are you filtering out here?

Reply:
the query is going to return target docs for other contacts that fit the interval. For a supervisor or an online user who have access to target docs from multiple contacts, this list is going to return roughly all target contacts within the 6 months period. 
Unfortunately, because the interval is first, there's no way to target the specific contact. 
I tried doing individual queries for every interval, but these version tested as best performing - an allDocs with a start and end key without docs, and then an allDocs with keys and include docs.

Reply:
Of course, sorry, this comment must have been pre-coffee. I read the start_key and end_key wrong...